service/src/main/java/org/cbioportal/service/impl/GenericAssayServiceImpl.java

```diff
@@ -58,16 +58,22 @@
                 List<String> stableIdsInMolecularProfiles = genericAssayRepository.getGenericAssayStableIdsByMolecularIds(distinctMolecularProfileIds);
                 allStableIds.addAll(stableIdsInMolecularProfiles);
             }
-        } 
-        if (stableIds != null) {
-			Map<String, String> allStableIdMap = allStableIds
-					.stream()
-					.collect(Collectors.toMap(stableId -> stableId, stableId -> stableId));
-
-			allStableIds = stableIds
-					.stream()
-					.filter(stableId -> allStableIdMap.containsKey(stableId))
-					.collect(Collectors.toSet());
+            // if stableIds and molecularProfileIds both exist, find the common
+            if (stableIds != null) {
+                Map<String, String> allStableIdMap = allStableIds
+                        .stream()
+                        .collect(Collectors.toMap(stableId -> stableId, stableId -> stableId));
```

Comment:
Not important, you could use identity function here
`.collect(Collectors.toMap(Function.identity(), Function.identity()));`